src/tools/base/segmentation/BaseCircleSegmentationTool.js

```diff
@@ -0,0 +1,171 @@
+import BaseSegmentationTool from '../BaseSegmentationTool';
+import toolColors from '../../../stateManagement/toolColors';
+import { draw, drawCircle, getNewContext } from '../../../drawing';
+import external from '../../../externalModules';
+import _isEmptyObject from './../../../util/isEmptyObject';
+import store from '../../../store';
+import { getRadius } from './../../../util/ellipse';
+
+const brushModule = store.modules.brush;
+const { getters } = brushModule;
+
+class BaseCircleSegmentationTool extends BaseSegmentationTool {
+  constructor(props, defaultProps = {}) {
+    if (!defaultProps.configuration) {
+      defaultProps.configuration = { alwaysEraseOnClick: false };
+    }
+
+    defaultProps.configuration.referencedToolData = 'segmentation';
+
+    super(props, defaultProps);
+  }
+
+  /**
+   * Render hook: draws the Scissors's outline, box, or circle
+   *
+   * @param {Object} evt Cornerstone.event#cornerstoneimagerendered > cornerstoneimagerendered event
+   * @memberof Tools.ScissorsTool
+   * @returns {void}
+   */
+  renderToolData(evt) {
+    const eventData = evt.detail;
+    const { element } = eventData;
+    const color = toolColors.getColorIfActive({ active: true });
+    const context = getNewContext(eventData.canvasContext.canvas);
+
+    draw(context, context => {
+      const startCanvas = external.cornerstone.pixelToCanvas(
+        element,
+        this.handles.start
+      );
+
+      const endCanvas = external.cornerstone.pixelToCanvas(
+        element,
+        this.handles.end
+      );
+
+      // Calculating the radius where startCanvas is the center of the circle to be drawn
+      const radius = getRadius(startCanvas, endCanvas);
```

Comment:
You don't need a new function for this, it's just the distance between two points. Remove this.

https://github.com/cornerstonejs/cornerstoneTools/blob/0aae680b20c3777866a49354465f5c2b78f7ffc6/src/tools/annotation/bidirectionalTool/moveHandle/perpendicularBothFixedRight.js#L39

https://github.com/cornerstonejs/cornerstoneTools/blob/0aae680b20c3777866a49354465f5c2b78f7ffc6/src/tools/annotation/bidirectionalTool/moveHandle/perpendicularBothFixedRight.js#L5

Reply:
Fixed.